apps/examples/testcase/le_tc/kernel/itc_environ.c

```diff
@@ -0,0 +1,253 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file itc_environ.c
+/// @brief Scenario Test Case for Environ API
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <errno.h>
+#include "tc_internal.h"
+
+#define OVERWRITE 1
+#define VAR "var"
+#define VALUE "value"
+#define VAR_COUNT 5
+#define BUFF_LEN 20
+
+/**
+* @fn                   :itc_environ_env_dup_env_findvar_p
+* @brief                :Copies the environment structure of a task, and finds the address of a environment variable by name
+* @scenario             :Find environment variable before and after duplicating
+* API's covered         :env_dup, env_findvar, setenv, env_release, clearenv
+* Preconditions         :none
+* Postconditions        :none
+*/
+static void itc_environ_env_dup_env_findvar_p(void)
+{
+	struct task_group_s group;
+	int i;
+	char var_buf[BUFF_LEN];
+	char value_buf[BUFF_LEN];
+	char *result;
+
+	clearenv();
+	for (i = 0; i < VAR_COUNT; i++) {
+		snprintf(var_buf, BUFF_LEN, "%s%d", VAR, i);
+		result = (char *)env_findvar(&group, var_buf);
+		TC_ASSERT_EQ("env_findvar", result, NULL);
+	}
+
+	TC_ASSERT_EQ("env_dup", env_dup(&group), OK);
+	for (i = 0; i < VAR_COUNT; i++) {
+		snprintf(var_buf, BUFF_LEN, "%s%d", VAR, i);
+		result = (char *)env_findvar(&group, var_buf);
+		TC_ASSERT_EQ("env_findvar", result, NULL);
+	}
+
+	for (i = 0; i < VAR_COUNT; i++) {
+		snprintf(var_buf, BUFF_LEN, "%s%d", VAR, i);
+		snprintf(value_buf, BUFF_LEN, "%s%d", VALUE, i);
+		TC_ASSERT_EQ_CLEANUP("setenv", setenv(var_buf, value_buf, OVERWRITE), OK, clearenv());
```

Comment:
I think clearenv() is not required in "ASSERT" in case of setenv. Please check for other testcases also.

Reply:
@arvinmittal it's done